openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -538,26 +538,43 @@ def run(self, *, dry=False, verbose=True,
             has_solvent=solvent_comp is not None,
         )
 
+        # workaround for conformer generation failures
+        # see openfe issue #576
+        # calculate partial charges manually if not already given
+        # convert to OpenFF here,
+        # and keep the molecule around to maintain the partial charges
+        off_small_mols = {
+            'stateA': mapping.componentA.to_openff(),
+            'stateB': mapping.componentB.to_openff(),
+            'both': [m.to_openff() for m in small_mols
+                     if (m != mapping.componentA and m != mapping.componentB)]
+        }
```

Comment:
from playing with TemplateGenerator, it didn't seem to cache the user charges I passed through (i.e. I couldn't convert to openff, assign partial charges, parametrise here to populate cache, then later retrieve the same partial charges from the SMILES lookup.).  Therefore I've converted the small molecules to openff here, and they remain in this representation throughout. 

I could probably put this dict representation in your previous validate functions

Reply:
> it didn't seem to cache the user charges I passed through

I'm re-reading this now - I find that super odd, I checked several times that SystemGenerator doesn't double calculate charges & will read from cache (did this by doing repeated independent executions of this code and checking for sqm calls on a big molecule).

I'm confused as to why this would be any different O.o! bug?

Reply:
Yeah I didn't mind the structure of AFE with each step being a separate block.  

re: the cache.  It was probably buggy behaviour for me locally, but it wasn't creating the .json file.  I think currently the use of cache is optional on our end too, so we can't rely on it either

Reply:
forgot it was optional 😅

Erm, I'll be honest I kinda feel like we should get this out to users now and bother about cleaning up later - how would you feel about going for option 2?